src/sonic-yang-models/yang-models/sonic-port.yang

```diff
@@ -110,9 +110,19 @@ module sonic-port{
 				}
 
 				leaf index {
-					type uint16 {
-						range 0..256;
+					type uint16;
+				}
+
+				leaf asic_port_name {
+					type string;
+					description "port name in asic and asic name";
+				}
+
+				leaf role {
```

Comment:
Should the role be valid only when asic_port_name is defined?

Reply:
@SuvarnaMeenakshi would you please confirm? Role is valid only when asic_port_name is defined?

Reply:
Yes, currently we only have role defined for asic and not for the device.
But as this field is not mandatory, defining it this way should be fine.
@arlakshm - please correct me if I am wrong.

Reply:
@dgsudharsan, @venkatmahalingam. The `role` is not related to `asic_port_name`. 
The `role` indicates if the port is an internal port or external(frontpanel) facing port. if a `role` is not specified then the port is assumed to be external facing
The 'asic_port_name` indicates the port on the packet processer. It is used to map Sonic port name to the port on the asic

IMO, these two are attributes not related and we should not have a condition that the role to  asic port_name and vice versa.